### ISSUE TITLE
[3.5] Could not determine country polygon containing the point.

### ISSUE THREAD

**Issue by Naahuel** [mediagis/nominatim-docker#297]
Hello! This might not be a bug, I'm just clueless as to what's happening and what I could do.

I have a very simple Linux server (Ubuntu 20.04.3 LTS (GNU/Linux 5.4.0-100-generic x86_64)). I'm using version 3.5 and I run it like this:

first I build the database:

```
DATA_PATH=$(realpath ../data)
mkdir $DATA_PATH
wget -O $DATA_PATH/data.osm.pbf https://download.openstreetmap.fr/extracts/south-america/argentina/tucuman.osm.pbf
cd nominatim-docker/3.5

# Build
docker build --pull --rm -t nominatim .

# Init db
docker run -t -v $DATA_PATH:/data nominatim  sh /app/init.sh /data/data.osm.pbf postgresdata 4
```

Then I run nominatim like this:

```
cd nominatim-docker/3.5
docker run \
    --restart=always \
    -p 6432:5432 \
    -p 7070:8080 \
    -v $DATA_PATH/postgresdata:/var/lib/postgresql/12/main \
    -d nominatim bash /app/start.sh
```

This has worked for like 2 years without any problems! One day out of the blue (I never touched it, updated it or anything), the reverse search stopped working and it ends with the error: `Could not determine country polygon containing the point.` (the PBF file I imported is very small, containing only a province/state)

using `debug=1` in the request: 
```
string(720) "select place_id,parent_place_id,rank_address,country_code, ST_distance(ST_SetSRID(ST_Point(-65.2067782,-26.834862),4326), geometry) as distance FROM  placex   WHERE ST_DWithin(ST_SetSRID(ST_Point(-65.2067782,-26.834862),4326), geometry, 0.006)   AND rank_address between 26 and 30 and (name is not null or housenumber is not null or rank_address between 26 and 27) and (rank_address between 26 and 27      or ST_GeometryType(geometry) != 'ST_LineString') and class not in ('boundary') and indexed_status = 0 and linked_place_id is null and (ST_GeometryType(geometry) not in ('ST_Polygon','ST_MultiPolygon')  OR ST_DWithin(ST_SetSRID(ST_Point(-65.2067782,-26.834862),4326), centroid, 0.006)) ORDER BY distance ASC limit 1"
bool(false)
{"error":{"code":500,"message":"Could not determine country polygon containing the point.","details":"/app/src/lib/DB.php(187)"}}
```

Of course when this happened I tried re-building everything (i purged all containers, images and volumes) but no luck, it always breaks like this now.

The search function doesn't throw an error, but it stopped working, it always returns an empty result.

I'm using the same PBF file to run openstreetmap-tile-server and osrm-backend and they work perfectly fine.

Any kind of help will be greatly appretiated! Thanks!

**Comment by philipkozeny** [comment]
Do you mind testing it with 4.0? It's way easier to use, just a single command and you don't even have to build the image by yourself. I have just tested it with:

```
docker run -it --rm \
  -e PBF_URL=https://download.openstreetmap.fr/extracts/south-america/argentina/tucuman.osm.pbf \
  -p 8080:8080 \
  --name nominatim \
  mediagis/nominatim:4.0
```

This downloads the image, grabs the `.osm.pbf` file, runs the import, etc. in one go.

Works fine in that case, I get results back when I query it. Also please read https://github.com/mediagis/nominatim-docker/blob/master/4.0/README.md if you want to make changes (like persisting the data of the Postgresql DB etc.)

**Comment by Naahuel** [comment]
Hello @philipkozeny thanks so much for responding!

I actually did! But gave up after a few tries of trying my script to reuse the PBF file and shared volume and got bunch of permission errors, then an Postgres memory error. I was just trying to make the old script I had work hehe. With that new implementation it is of course 10x more convenient and easy! So since the pbf file i'm using is extremely small, I guess it's fine downloading it twice, who cares.

As side notes: I did have to reduce POSTGRES_SHARED_BUFFERS to 1GB for some reason (the memory error I mentioned above).

The reverse request faled at first (not even a server error, just `ERR_EMPTY_RESPONSE`.)
![image](https://user-images.githubusercontent.com/1612488/155124070-6477d794-dedf-4828-8212-580c71e0e679.png)

The WEIRD thing is that if I pass `debug=1` to the request, it worked :confused: 
![image](https://user-images.githubusercontent.com/1612488/155124527-367f1960-4e3c-46eb-b2cb-5024e85cc8b4.png)

So after trying many things, I modified a bash script I had to run all the containers in my server and I used detached mode for nominatim 4.0 and now it works fine :shrug: 

Anyway, thanks for your help! I wish I knew docker better (or nominatim...) to help you debug my original problem, but if it turns out to be a bug and there's any extra info you need just let me know!

**Comment by Naahuel** [comment]
Thanks @philipkozeny 

Yeah, I got it working (sorry if that wasn't clear in my previous post). I ran it on the exact same server as in my original post. 

The only thing that i changed to make it run is use detached mode. Now it works just fine.

I'm sure it was something wrong with my setup process. If it happens again, i'll see if I can recover some logs, in case it's useful.

Thanks so much again, you've been of great help!
